main.go

```diff
@@ -136,6 +138,53 @@ func (ac *AtomicCounter) Next() int64 {
 	return atomic.AddInt64(&ac.val, 1)
 }
 
+// Bare minimum auth middleware in lieu of major restructuring.
+// Assumes we are reading a K8S secret mounted with the other secrets
+// read in by the init() func at /etc/assist-secret/assist-token
+// This can be prepopulated by the mk_password.py file.
+func AuthMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// Ensure that we indicate authorization may vary
+		w.Header().Add("Vary", "Authorization")
+
+		// Returns "" empty string if nothing is found.
+		authorizationHeader := r.Header.Get("Authorization")
+
+		// If no auth header, set user as anonymous
+		if authorizationHeader == "" {
+			w.Header().Set("WWW-Authenticate", "Bearer")
+			message := "invalid or missing auth token"
+			http.Error(w, message, http.StatusUnauthorized)
+			return
+		}
+
+		headerParts := strings.Split(authorizationHeader, " ")
+		if len(headerParts) != 2 || headerParts[0] != "Bearer" {
+			w.Header().Set("WWW-Authenticate", "Bearer")
+			message := "invalid or missing auth token"
+			http.Error(w, message, http.StatusUnauthorized)
+			return
+		}
+
+		token := headerParts[1]
+
+		admin, err := os.ReadFile("/etc/assist-secret/assist-token")
```

Comment:
should maybe be read in getaccess for consistency

Reply:
That's fine, I've updated to use a global variable and added another init function to initialize it (similar to how getaccess,etc is done). None of the code in here is consistent though, it needs a few best practice changes but for now it works as is.